src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/ClassicalControl.qs

```diff
@@ -857,4 +822,28 @@ namespace Microsoft.Quantum.Testing.ClassicalControl {
         controlled adjoint self;
     }
 
+}
+
+// =================================
+
+namespace Microsoft.Quantum.Testing.ClassicalControl {
+    open SubOps;
+
+    // Within Block Support
+
+    operation Foo() : Unit {
+        let r = One;
+        within {
+            if (r == Zero) {
+                SubOp1();
+                SubOp2();
+            }
+        } apply {
+            if (r == One) {

```

Comment:
It will follow the standard rules for applying characteristics to hoisted content. That does not always mean having the same characteristics as its parent.

Reply:
Rephrasing: It has at least the same support as Foo, correct?